File: app/src/main/java/com/hfad/myferma/incubator/HomeAdapterIncubator.java
package com.hfad.myferma.incubator;

import android.annotation.SuppressLint;
import android.graphics.drawable.Drawable;
import android.os.Build;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.RequiresApi;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.card.MaterialCardView;
import com.hfad.myferma.R;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Calendar;
import java.util.Date;
import java.util.concurrent.TimeUnit;

public class HomeAdapterIncubator extends RecyclerView.Adapter<HomeAdapterIncubator.ViewHolder> {
    private ArrayList id, name, type, data, arhive;
    private long diff;
    private Listener listener;

    public static interface Listener {
        public void onClick(int position, String name, String type, String data, String id);
    }

    public HomeAdapterIncubator(ArrayList idIncubator, ArrayList nameIncubator, ArrayList typeIncubator, ArrayList dataIncubator, ArrayList arhiveIncubator) {
        this.id = idIncubator;
        this.name = nameIncubator;
        this.type = typeIncubator;
        this.data = dataIncubator;
        this.arhive = arhiveIncubator;
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        private MaterialCardView cardView;

        public ViewHolder(MaterialCardView v) {
            super(v);
            cardView = v;
        }
    }

    @NonNull
    @Override
    public HomeAdapterIncubator.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        MaterialCardView cv = (MaterialCardView) LayoutInflater.from(parent.getContext())
                .inflate(R.layout.card_captioned_image, parent, false);
        return new ViewHolder(cv);
    }

    public void setListener(Listener listener) {
        this.listener = listener;
    }

    @RequiresApi(api = Build.VERSION_CODES.O)
    @Override
    public void onBindViewHolder(@NonNull HomeAdapterIncubator.ViewHolder holder, @SuppressLint("RecyclerView") final int position) {

        MaterialCardView cardView = holder.cardView;

        ImageView imageView = (ImageView) cardView.findViewById(R.id.info_image);
        //Установка картинки в карту
        Drawable drawable = null;
        if ("Курицы".equals(type.get(position))) {
            drawable = cardView.getResources().getDrawable(R.drawable.chicken);
        } else if ("Гуси".equals(type.get(position))) {
            drawable = cardView.getResources().getDrawable(R.drawable.external_goose_birds_icongeek26_outline_icongeek26);
        } else if ("Перепела".equals(type.get(position))) {
            drawable = cardView.getResources().getDrawable(R.drawable.quail);
        } else if ("Утки".equals(type.get(position))) {
            drawable = cardView.getResources().getDrawable(R.drawable.duck);
        } else if ("Индюки".equals(type.get(position))) {
            drawable = cardView.getResources().getDrawable(R.drawable.turkeycock);
        }

        imageView.setImageDrawable(drawable);
        imageView.setContentDescription("22");

        Calendar calendar = Calendar.getInstance();
        String dateBefore22 = String.valueOf(data.get(position));
        String dateBefore222 = calendar.get(Calendar.DAY_OF_MONTH) + 1 + "." + (calendar.get(Calendar.MONTH) + 1) + "." + calendar.get(Calendar.YEAR);

        SimpleDateFormat myFormat = new SimpleDateFormat("dd.MM.yyyy");

        try {
            Date date1 = myFormat.parse(dateBefore22);
            Date date2 = myFormat.parse(dateBefore222);
            diff = date2.getTime() - date1.getTime();
        } catch (ParseException e) {
            throw new RuntimeException(e);
        }

        TextView textView = (TextView) cardView.findViewById(R.id.name);
        textView.setText(String.valueOf(name.get(position)));// имя инкубатора
        TextView textView1 = (TextView) cardView.findViewById(R.id.dayEnd);
        textView1.setText("Идет " + String.valueOf(TimeUnit.DAYS.convert(diff, TimeUnit.MILLISECONDS)) + " день ");//Какой день
        cardView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (listener != null) {
                    listener.onClick(position, String.valueOf(name.get(position)), String.valueOf(type.get(position)), String.valueOf(data.get(position)), String.valueOf(id.get(position)));
                }
            }
        });

    }

    @Override
    public int getItemCount() {
        return id.size();
    }

}

File: app/src/main/java/com/hfad/myferma/incubator/MenuIncubator/HomeAdapterIncubator.kt
package com.hfad.myferma.incubator.MenuIncubators

import android.graphics.drawable.Drawable
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.card.MaterialCardView
import com.hfad.myferma.R
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.Calendar
import java.util.Date
import java.util.concurrent.TimeUnit


class HomeAdapterIncubator constructor(
    private val id: List<String>,
    private val name: List<String>,
    private val type: List<String>,
    private val data: List<String>, private val listener: Listener, private val home: Boolean
) : RecyclerView.Adapter<HomeAdapterIncubator.ViewHolder?>() {


    interface Listener {
        fun onClick(position: Int, name: String, type: String, data: String, id: String)
    }

    class ViewHolder constructor(val cardView: MaterialCardView) : RecyclerView.ViewHolder(
        cardView
    )

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        val cv: MaterialCardView = LayoutInflater.from(parent.context)
            .inflate(R.layout.card_captioned_image, parent, false) as MaterialCardView
        return ViewHolder(cv)
    }

    override fun onBindViewHolder(
        holder: ViewHolder, position: Int
    ) {
        val cardView: MaterialCardView = holder.cardView
        val imageView: ImageView = cardView.findViewById<View>(R.id.info_image) as ImageView
        var diff: Long = 0

        //Установка картинки в карту
        var drawable: Drawable? = null
        if ("Курицы" == type[position]) {
            drawable = cardView.resources.getDrawable(R.drawable.chicken)
        } else if ("Гуси" == type[position]) {
            drawable = cardView.resources
                .getDrawable(R.drawable.external_goose_birds_icongeek26_outline_icongeek26)
        } else if ("Перепела" == type[position]) {
            drawable = cardView.resources.getDrawable(R.drawable.quail)
        } else if ("Утки" == type[position]) {
            drawable = cardView.resources.getDrawable(R.drawable.duck)
        } else if ("Индюки" == type[position]) {
            drawable = cardView.resources.getDrawable(R.drawable.turkeycock)
        }
        imageView.setImageDrawable(drawable)
        imageView.contentDescription = "22"

        var dateIncubator = ""
        if (home) {
            val calendar: Calendar = Calendar.getInstance()
            val dateBefore22: String = data[position].toString()
            val dateBefore222: String =
                (calendar.get(Calendar.DAY_OF_MONTH) + 1).toString() + "." + (calendar.get(
                    Calendar.MONTH
                ) + 1) + "." + calendar.get(Calendar.YEAR)
            val myFormat: SimpleDateFormat = SimpleDateFormat("dd.MM.yyyy")
            try {
                val date1: Date = myFormat.parse(dateBefore22)
                val date2: Date = myFormat.parse(dateBefore222)
                diff = date2.time - date1.time
            } catch (e: ParseException) {
                throw RuntimeException(e)
            }
            dateIncubator =  "Идет " + TimeUnit.DAYS.convert(diff, TimeUnit.MILLISECONDS).toString() + " день "
        } else {
            dateIncubator = "Завершено"
        }
        val textView: TextView = cardView.findViewById<View>(R.id.name) as TextView
        textView.text = name[position] // имя инкубатора

        val textView1: TextView = cardView.findViewById<View>(R.id.dayEnd) as TextView
        textView1.text = dateIncubator //Какой день

        cardView.setOnClickListener {
            listener.onClick(
                position,
                name[position],
                type[position],
                data[position],
                id[position]
            )
        }
    }

    override fun getItemCount(): Int {
        return id.size
    }
}